src/test/regress/sql/alter_role_propagation.sql

```diff
@@ -78,11 +78,66 @@ SELECT row(rolname, rolsuper, rolinherit,  rolcreaterole, rolcreatedb, rolcanlog
 SELECT run_command_on_workers($$SELECT row(rolname, rolsuper, rolinherit,  rolcreaterole, rolcreatedb, rolcanlogin, rolreplication, rolbypassrls, rolconnlimit, rolpassword, EXTRACT (year FROM rolvaliduntil)) FROM pg_authid WHERE rolname = 'alter_role_1'$$);
 
 
--- table belongs to a role
+-- give login permissions so that we can connect and check if the previous queries were propagated
+ALTER ROLE alter_role_1 LOGIN CONNECTION LIMIT 10;
 
 
--- we don't support propagation of configuration_parameters and notice the users
+-- alter configuration_parameter defaults for a user
 ALTER ROLE alter_role_1 SET enable_hashagg TO FALSE;
+ALTER ROLE alter_role_1 IN DATABASE template1 SET enable_hashjoin TO 0;
+
+-- as the defaults take effect on new connections, reconnect as the test user
+\c - alter_role_1
+SHOW enable_hashagg;
+SHOW enable_hashjoin;
+
+\c template1
+SHOW enable_hashagg;
+SHOW enable_hashjoin;
+
+-- verify that the config defaults are propagated in the workers
+\c postgres - - :worker_1_port
```

Comment:
It would be nice to avoid `\c` and use `run_command_on_workers`. Main reason is that we want to start running a subset of our tests against Hyperscale and \c will not work there.

Reply:
Is there an easy way to run `run_command_on_workers` with a given user, and in a seperate database?

Some problems that I ran into that made me choose `\c` over `run_command_on_workers`:
* I was unable to connect to **separate databases** using `run_command_on_workers` ( When I want to test `ALTER USER u IN DATABASE SET ..` queries)
* I was unable to read **session defaults for a different role** when I set it with `SET ROLE` (See `SET ROLE does not process session variables as specified by the role's ALTER ROLE settings; this only happens during login.` in https://www.postgresql.org/docs/12/sql-set-role.html notes section)

Reply:
I think just using the current database name and username for those tests gives pretty decent coverage.  Alternatively, you could check the catalog tables (see other tests in this file).

Reply:
Even better alternative could be to extract the happy ALTER ROLE..SET cases into a separate test file that does not rely on \c and leave the tricky ones here. 

Looking a bit more at this test file, it is unlikely to work on Hyperscale because we cannot create roles directly from the coordinator on Hyperscale.

Reply:
I did not introduce any new databases, or roles in this file. I had a chat with Ozan, and we decided it is not creating any new complications in this test.